src/store/base.rs

```diff
@@ -0,0 +1,210 @@
+    }
+
+    pub fn get_all_pacbuilds_by_name_like_and_install_state(
+        &self,
+        name_like: &str,
+        install_state: InstallState,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(Some(name_like), Some(install_state), None, None)
+    }
+
+    pub fn get_all_pacbuilds_by_name_like_and_repository_url(
+        &self,
+        name_like: &str,
+        url: &str,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(Some(name_like), None, None, Some(url))
+    }
+
+    pub fn get_all_pacbuilds_by_name_like_and_install_state_and_kind(
+        &self,
+        name_like: &str,
+        install_state: InstallState,
+        kind: Kind,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(Some(name_like), Some(install_state), Some(kind), None)
+    }
+
+    pub fn get_all_pacbuilds_by_name_like_and_install_state_and_repository_url(
+        &self,
+        name_like: &str,
+        install_state: InstallState,
+        url: &str,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(Some(name_like), Some(install_state), None, Some(url))
+    }
+
+    pub fn get_all_pacbuilds_by_name_like_and_install_state_and_kind_and_repository_url(
+        &self,
+        name_like: &str,
+        install_state: InstallState,
+        kind: Kind,
+        url: &str,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(Some(name_like), Some(install_state), Some(kind), Some(url))
+    }
+
+    pub fn get_all_pacbuilds_by_kind(&self, kind: Kind) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(None, None, Some(kind), None)
+    }
+
+    pub fn get_all_pacbuilds_by_kind_and_install_state(
+        &self,
+        kind: Kind,
+        install_state: InstallState,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(None, Some(install_state), Some(kind), None)
+    }
+
+    pub fn get_all_pacbuilds_by_kind_and_repository_url(
+        &self,
+        kind: Kind,
+        url: &str,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(None, None, Some(kind), Some(url))
+    }
+
+    pub fn get_all_pacbuilds_by_kind_and_install_state_and_repository_url(
+        &self,
+        kind: Kind,
+        install_state: InstallState,
+        url: &str,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(None, Some(install_state), Some(kind), Some(url))
+    }
+
+    pub fn get_all_pacbuilds_by_install_state(
+        &self,
+        install_state: InstallState,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(None, Some(install_state), None, None)
+    }
+
+    pub fn get_all_pacbuilds_by_install_state_and_repository_url(
+        &self,
+        install_state: InstallState,
+        url: &str,
+    ) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(None, Some(install_state), None, Some(url))
+    }
+
+    pub fn get_all_pacbuilds_by_repository_url(&self, url: &str) -> Vec<&PacBuild> {
+        self.get_all_pacbuilds_by(None, None, None, Some(url))
+    }
+}
```

Comment:
Instead of creating so many methods, I would suggest to use the Builder Pattern.

Here's a video explaining the pattern in Rust: https://www.youtube.com/watch?v=5DWU-56mjmg